### PR TITLE
Allowing moderators to add WCW Topics

### PR DESCRIPTION
**Pull request by DarrylBrooks97**
[Demo](https://www.loom.com/share/783da610f2c64f7181291a8cc8effc9b)

### CODE REVIEW

File: src/commands/update-wcw.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    .setDescription('Add a topic for Water Cooler Wednesdays 💦')
+    .addStringOption(option =>
+        option.setName('topic').setDescription('Your topic').setRequired(true),
+    );
+
+export async function execute(interaction: CommandInteraction<CacheType>) {
+    const topic = interaction.options.getString('topic') as string;
+    const interactionAuthor = interaction.member?.user as User;
+
+    if (!CONSTANTS.MODERATORS.includes(interactionAuthor.id)) {
```
Comment (blaccsmith) [inline src/commands/update-wcw.ts]: This isn't maintainable long term. Instead, you can check if the author's roles contains the Moderator role. This means we won't have to redeploy this code whenever Mods change.

File: src/commands/update-wcw.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        await Promise.allSettled([
```
Comment (blaccsmith) [inline src/commands/update-wcw.ts]: Simple `await` this since there's only one action happening

File: src/constants.ts [hunk 3]
```diff
@@ -20,6 +20,7 @@ export const CONSTANTS = Object.freeze({
     BOT_PARTIALS: ['MESSAGE', 'CHANNEL', 'REACTION'] as PartialTypes[],
     SPECTATOR_ROLE_ID: process.env.SPECTATOR_ROLE_ID as string,
     MEMBER_ROLE_ID: process.env.MEMBER_ROLE_ID as string,
+    MODERATORS: [process.env.GARY_ID, process.env.GARRETT_ID, process.env.DARRYL_ID],
```
Comment (blaccsmith) [inline src/constants.ts]: Not needed – see first comment.
```suggestion
```

**Review by blaccsmith — CHANGES_REQUESTED** [review]

**Comment by Genyus** [comment]
```diff
diff --git a/scripts/sync-kinsta.sh b/scripts/sync-kinsta.sh
index 37f2cd3..c872b68 100755
--- a/scripts/sync-kinsta.sh
+++ b/scripts/sync-kinsta.sh
@@ -20,7 +20,7 @@ declare -A DEV=(
        ["rootdomain"]="frocentric.local"
        ["domain"]="frocentric.local"
        ["url"]="https://frocentric.local"
-       ["discourseurl"]="https://localhost:4200"
+       ["discourseurl"]="https://tech-community.staging.frocentric.io"
 )
 declare -A STAGING=(
        ["bucket"]="s3://froware-staging/uploads"
```

File: src/commands/update-wcw.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        logger({
+            project: 'blacc',
+            channel: 'general',
+            event: 'WCW topic added',
+            description: member.id,
+            icon: '🟢',
+            notify: true,
+        }),
+    ]);
+}
```
Comment (Genyus) [inline src/commands/update-wcw.ts]: Don't need to hold up this PR, but would suggest adding the ability to list and remove/edit topics, plus checking for duplicates when creating new ones

File: src/lib/addTopic.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { PrismaClient } from '@prisma/client';
+
+export const addTopic = async (topic: string) => {
+    const prisma = new PrismaClient();
+    return await prisma.wcwTopic.create({
+        data: {
+            used: false,
+            topic,
+        },
+    });
```
Comment (Genyus) [inline src/lib/addTopic.ts:10]: Does this need any error checking?

File: src/prisma/schema.prisma [hunk 6]
```diff
@@ -147,4 +147,4 @@ model wcwTopic {
   used  Boolean
   createdAt DateTime      @default(now())
   updatedAt DateTime      @updatedAt
-}
+
```
Comment (Genyus) [inline src/prisma/schema.prisma]: This looks like an accidental, breaking change?

**Review by Genyus — CHANGES_REQUESTED** [review]
Looks like prisma.schema needs to be reverted

File: src/lib/addTopic.ts (hunk 5 above)
Comment (gt-codes) [reply src/lib/addTopic.ts:10]: In this case, no.

File: src/prisma/schema.prisma (hunk 6 above)
Comment (gt-codes) [reply src/prisma/schema.prisma]: Thanks for catching this! I pushed a fix

File: src/commands/update-wcw.ts (hunk 4 above)
Comment (gt-codes) [reply src/commands/update-wcw.ts]: Just pushed these request

File: src/commands/intro.ts [hunk 7]
```diff
@@ -92,7 +92,7 @@ export async function execute(interaction: CommandInteraction<CacheType>) {
             channel: 'welcome',
             event: 'New introduction',
             description: member.user.tag,
-            icon: '🟢',
+            icon: '👋',
```
Comment (Genyus) [inline src/commands/intro.ts:95]: Can we get a skin-tone, at least? 👋🏾

**Review by Genyus — APPROVED** [review]
LGTM

**Review by blaccsmith — APPROVED** [review]

**Pull request merged**